soc/nordic/common/uicr/gen_uicr.py

```diff
@@ -186,48 +189,43 @@ def main() -> None:
         uicr.VERSION.MAJOR = UICR_FORMAT_VERSION_MAJOR
         uicr.VERSION.MINOR = UICR_FORMAT_VERSION_MINOR
 
-        kconfig_str = args.in_config.read()
-        kconfig = parse_kconfig(kconfig_str)
+        # Create separate hex objects
+        merged_hex = IntelHex()
+        uicr_hex = IntelHex()
+        periphconf_hex = IntelHex()
 
-        edt = pickle.load(args.in_edt_pickle)
+        # Always add UICR data
+        uicr_data = bytes(uicr)
+        uicr_hex.frombytes(uicr_data, offset=args.uicr_address)
+        merged_hex.frombytes(uicr_data, offset=args.uicr_address)
 
-        try:
-            periphconf_partition = edt.label2node[PERIPHCONF_NODELABEL]
-        except LookupError as e:
-            raise ScriptError(
-                "Failed to find a PERIPHCONF partition in the devicetree. "
-                f"Expected a DT node with label '{PERIPHCONF_NODELABEL}'."
-            ) from e
+        if args.in_periphconf_elfs:  # Check if periphconf data is provided
+            periphconf_combined = extract_and_combine_periphconfs(args.in_periphconf_elfs)
```

Comment:
I think we should retain the ability to generate an empty (0xFFF_FFFFs) periphconf blob if no elf is provided here - an empty periphconf is still valid

Reply:
fixed